### PR TITLE
Allow integer response code

### LINKED ISSUE #26: schema 2.0 return expected string or buffer
Without specifying `--schema 2.0` the file is read, but a parsing error occurs (see bottom), specifying 2.0 returns `expected string or buffer`. This should be a 2.0 valid file, in use with connexxion.

Is this due to an error in parsing the yaml, or is something not right when using this parameter?

```
openapi-spec-validator --schema 2.0 cms/cms/openapi/cms.yaml
expected string or buffer
```

```
'openapi' is a required property

Failed validating 'required' in schema:
```

### PR DESCRIPTION
**Pull request by p1c2u**
This PR relates to #29 

Introduces extended yaml safe loader which converts integer keys to strings 

Fixes #26

### CODE REVIEW

File: openapi_spec_validator/handlers.py [hunk 1]
```diff
@@ -3,17 +3,20 @@
...
+from yaml import load
+
+from openapi_spec_validator.loaders import ExtendedSafeLoader
 
 
 class UrlHandler:
     """OpenAPI spec validator URL scheme handler."""
 
-    def __init__(self, *allowed_schemes):
+    def __init__(self, *allowed_schemes, loader=ExtendedSafeLoader):
```
Comment (dtkav) [inline openapi_spec_validator/handlers.py]: looks like this syntax breaks python2.7. maybe allow overriding the loader with `**kwargs` instead?

Comment (p1c2u) [reply openapi_spec_validator/handlers.py]: Yup, hat's the only way to support py2

**Comment by p1c2u** [comment]
@dtkav @gcallaghan @TomFaulkner @erunion That should solve the issue. Thanks for thelp guys.

**Comment by jmagnusson** [comment]
This problem still seems to be here, even when using `openapi-spec-validator==0.2.4`:

```
❯ pip install PyYAML==3.13 openapi-spec-validator==0.2.4; python -c "import yaml
import openapi_spec_validator
d = yaml.load('''openapi: '3.0.0'
info:
  title: Minimal valid OpenAPI specification
  version: '0.1'
paths:
  /status:
    get:
      responses:
        200:
          description: Return the API status.''')
openapi_spec_validator.validate_spec(d)"

Requirement already satisfied: PyYAML==3.13 in /usr/local/lib/python3.6/site-packages (3.13)
Requirement already satisfied: openapi-spec-validator==0.2.4 in /usr/local/lib/python3.6/site-packages (0.2.4)
Requirement already satisfied: six in /usr/local/lib/python3.6/site-packages (from openapi-spec-validator==0.2.4) (1.11.0)
Requirement already satisfied: jsonschema in /usr/local/lib/python3.6/site-packages (from openapi-spec-validator==0.2.4) (2.6.0)

Traceback (most recent call last):
  File "<string>", line 13, in <module>
  File "/usr/local/lib/python3.6/site-packages/openapi_spec_validator/shortcuts.py", line 7, in validate
    return validator_callable(spec, spec_url=spec_url)
[...SNIP...]
  File "/usr/local/lib/python3.6/site-packages/jsonschema/_validators.py", line 25, in additionalProperties
    extras = set(_utils.find_additional_properties(instance, schema))
  File "/usr/local/lib/python3.6/site-packages/jsonschema/_utils.py", line 104, in find_additional_properties
    if patterns and re.search(patterns, property):
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/re.py", line 182, in search
    return _compile(pattern, flags).search(string)
TypeError: expected string or bytes-like object
```

**Pull request merged**